### PR TITLE
[Rootchain Staking] Add rootchain config command

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR adds a rootchain configuration, and a command interface for setting that configuration which other Rootchain Staking modules will use.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

Manual tests with feeding the correct ommand input / wrong command input.

# Documentation update
// TODO

# Additional comments

Fixes EDGE-817

### CODE REVIEW

File: command/rootchain/params.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	errUnableToReadConfig = errors.New("unable to read rootchain config")
+)
+
+type rootchainParams struct {
+	configPath    string
+	rootchainAddr string
+	eventABI      string
+	methodABI     string
+	localAddr     string
+	payloadType   uint64
```
Comment (0xAleksaOpacic) [inline command/rootchain/params.go]: Just to get a context, what is this for? If it's not required we should add in command description that default is 0 :)

File: command/rootchain/params.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+func (p *rootchainParams) validateMethodABI() error {
+	_, err := abi.NewABI(p.methodABI)
+	if err != nil {
+		return errInvalidMethodABI
+	}
+
+	return nil
+}
+
+func (p *rootchainParams) validateLocalAddress() error {
```
Comment (0xAleksaOpacic) [inline command/rootchain/params.go:110]: Do we need to validate that local-addr and rootchain-addr are of type.Address, that the format is good?

**Review by 0xAleksaOpacic — APPROVED** [review]
Let's hope every effort for this feature goes this smoothly, LGTM.

I haven't tested this manually, can you please add example files and commands to i can try it locally

File: command/rootchain/params.go (hunk 2 above)
Comment (zivkovicmilos) [reply command/rootchain/params.go:110]: The rootchain address is not of type `types.Address`, but a WS endpoint.
The local-addr value is of type `types.Address`.

I don't see how we can check the validity of the local address, apart from just making sure it's set ("0x1" is a valid address that will get expanded for example)

Comment (0xAleksaOpacic) [reply command/rootchain/params.go:110]: kk :D

File: command/rootchain/params.go (hunk 1 above)
Comment (zivkovicmilos) [reply command/rootchain/params.go]: It's used for packing / unpacking specific event payloads. Right now we only have a single value - `ValidatorSetPayload`, that's by default 0 (that's why it's not showing up)

**Comment by zivkovicmilos** [comment]
@Aleksao998 

You can try running for example:
`polygon-edge rootchain
--event-abi
"event StateSynced(uint256 indexed id, address indexed contractAddress, bytes data)"
--local-addr
"0x16Dc086Be70527bBB14E69A38292917F127C69b7"
--payload-type
0
--rootchain-addr
"ws://ethereum.org/ws"
--method-abi
"[ { \"anonymous\": false, \"inputs\": [ { \"indexed\": false, \"internalType\": \"bytes\", \"name\": \"data\", \"type\": \"bytes\" } ], \"name\": \"StateReceived\", \"type\": \"event\" }, { \"inputs\": [ { \"internalType\": \"bytes\", \"name\": \"data\", \"type\": \"bytes\" } ], \"name\": \"onStateReceived\", \"outputs\": [], \"stateMutability\": \"nonpayable\", \"type\": \"function\" } ]"`

File: command/rootchain/params.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	payloadTypeFlag        = "payload-type"
+	blockConfirmationsFlag = "block-confirmations"
+)
+
+var (
+	params = &rootchainParams{}
+)
+
+const (
+	defaultRootchainName = "rootchain.json"
```
Comment (0xAleksaOpacic) [inline command/rootchain/params.go:31]: Does it make sense to put this in .gitignore as we did for other json files

File: command/rootchain/rootchain.go [hunk 4]
```diff
@@ -66,6 +66,13 @@ func setFlags(cmd *cobra.Command) {
 		0,
 		"the payload type for the rootchain event. Possible types: [0 - ValidatorSetPayload]",
 	)
+
+	cmd.Flags().Uint64Var(
+		&params.blockConfirmations,
+		blockConfirmationsFlag,
+		0,
```
Comment (0xAleksaOpacic) [inline command/rootchain/rootchain.go]: I would put a default value as a bigger number (min 6)

Comment (zivkovicmilos) [reply command/rootchain/rootchain.go]: Why? It's `iota`

File: command/rootchain/params.go (hunk 3 above)
Comment (zivkovicmilos) [reply command/rootchain/params.go:31]: Added 👍 

[ad5dcff](https://github.com/0xPolygon/polygon-edge/pull/716/commits/ad5dcff42668045b0d03607e7f0399885b7b4d38)

File: command/rootchain/params.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	}
+
+	return nil
+}
+
+func (p *rootchainParams) loadExistingConfig() error {
+	_, err := os.Stat(p.configPath)
+	if err != nil && !os.IsNotExist(err) {
+		return errors.New("unable to stat rootchain config")
+	}
```
Comment (dbrajovic) [inline command/rootchain/params.go:122]: Short if ?

**Review by dbrajovic — APPROVED** [review]

File: command/rootchain/rootchain.go (hunk 4 above)
Comment (zivkovicmilos) [reply command/rootchain/rootchain.go]: I'm obviously blind, I thought you were talking about the payload type 🤦 

Bumped:
[1ac3b9e](https://github.com/0xPolygon/polygon-edge/pull/716/commits/1ac3b9e09d02d5bd5bd80941c99822368f166c24)

**Pull request merged**